core/state_transition.go

```diff
@@ -405,6 +418,20 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	}
 	st.gasRemaining -= gas
 
+	if rules.IsEIP4762 {
+		targetAddr := msg.To
+		originAddr := msg.From
+
+		st.evm.AccessEvents.AddTxOrigin(originAddr.Bytes())
+
+		if msg.To != nil {
+			st.evm.AccessEvents.AddTxDestination(targetAddr.Bytes(), msg.Value.Sign() != 0)
+
+			// ensure the code size ends up in the access witness
+			st.evm.StateDB.GetCodeSize(*targetAddr)
```

Comment:
Any particular reason to call `st.evm.StateDB.GetCodeSize(*targetAddr)`?

The codeSize will be accessed in `AddTxDestination` right? And also the `AccessEvent` is just for gas accounting, it doesn't make sense to raise a `GetCodeSize` call here iiuc.